src/meta/typesmap.go

```diff
@@ -72,30 +86,32 @@ func (m TypesMap) IsResolved() bool {
 
 // NewEmptyTypesMap creates new type map that has no types in it
 func NewEmptyTypesMap(cap int) TypesMap {
-	return TypesMap{m: make(map[string]struct{}, cap)}
+	return TypesMap{m: make(map[Type]struct{}, cap)}
 }
 
-func NewTypesMapFromTypes(types []Type) TypesMap {
-	m := make(map[string]struct{}, len(types))
-	for _, typ := range types {
-		s := typ.Elem
-		for i := 0; i < typ.Dims; i++ {
-			s = WrapArrayOf(s)
-		}
-		m[s] = struct{}{}
+func NewTypesMapFromPhpDocTypes(phpDocTypes []PhpDocType) TypesMap {
+	m := make(RawTypesMap, len(phpDocTypes))
```

Comment:
I would suggest to introduce New() method here instead of making an arbitrary type...